src/testdrive.F90

```diff
@@ -1969,4 +1971,240 @@ end function is_nan_qp
 #endif
 
 
+  subroutine error_wrap(error, more)
+    !> Error handling
+    type(error_type), intent(inout) :: error
+    !> Error message
+    character(len=*), intent(in) :: more
+
+    character(len=*), parameter :: skip = new_line("a") // repeat(" ", 11)
+
+    error%message = error%message // skip // more
+
+  end subroutine error_wrap
+
+
+  subroutine check_single_array(error, array, message, more)
+
+    !> Error handing
+    type(error_type), allocatable, intent(out) :: error
+
+    !> The array to be checked
+    class(*), intent(in), target :: array(:)
+
+    !> A detailed message describing the error
+    character(len=*), intent(in), optional :: message
+
+    !> Another line of error message
+    character(len=*), intent(in), optional :: more
+
+    integer :: i
+    class(*), pointer :: item(:) ! @note gfortran <=10 does not support syntax: associate(item => array(i))
+
+    item => array
+    do i = 1, size(array)
+      select type (item)
+      type is (integer)
```

Comment:
I don't know if you would agree to add `fypp` in this project, but I think it could be helpful for such a structure.

Reply:
If it can be made work with all three build system (fpm, meson and CMake), I would be fine, however there is currently no way to support this intrinsically with fpm.